### DISCUSSION TITLE
NGAlert - PagerDuty Notifier

### DISCUSSION THREAD

**Discussion by ssimard24** [grafana/grafana#47660] [Alerting]
In the old PagerDuty notifier module there was support for using Labels to dynamically update some PagerDuty event properties like: severity, dedup_key, summary, component, etc.

This functionality is now non-existent in the new PagerDuty alert notifier. 
Any plan to add this back in?
Can we use the old notifier with the new Grafana Alerts?

I am trying to achieve an alert where a single alert will be able to raise issues for a multitude of different microservices. This can only be possible if I can dynamically set some of the PagerDuty event properties from the Alert data. 
I also desire to use the PagerDuty Event Orchestration to control where the alert is forwarded. Isolating that knowledge in pagerduty offers a good separation of knowledge. The alert info now only needs to include the Component (micro service name) and the PagerDuty Event Orchestration key. PagerDuty does the routing work.

Old: https://github.com/grafana/grafana/blob/v8.1.2/pkg/services/alerting/notifiers/pagerduty.go
New: https://github.com/grafana/grafana/blob/v8.1.2/pkg/services/ngalert/notifier/channels/pagerduty.go

**Comment by trexx** [comment]
I feel like the PagerDuty provider is not really at feature parity with some of the other notifiers.
I'd love to be able to use message templates to customise the custom_details payload of the notification. Its currently hardcoded, and the current values cause a mess in the incident and require a lot of cleanup.

**Reply by trexx** [reply]
@yuri-tceretian Thanks so much for your recent changes to the PagerDuty notifier. I think they go a long way to improving its flexibility when it comes to customising the alert going to PD.
I am hoping you could also consider allowing us to customise the customdetails / message payload of the alert. In comparison to other notifiers where the message can be templatable, we don't seem to have that option in PD. Such change seems to be similar to your recent improvements to the notifier and I hope my suggestion is somewhat in scope.

**Comment by ssimard24** [comment]
Hi trexx. There is a related bug you can track: https://github.com/grafana/grafana/issues/48607

**Reply by trexx** [reply]
Thanks but that seems to be mostly a UI change as the configuration already allows for those fields to be templatable. Unlike custom_details where we're not able to make use of message templates since the PD notifier has a hard coded template. If we compare it to the Slack notifier, message body is an optional templatable configuration option in the notifier.

https://github.com/grafana/grafana/blob/v9.3.0-beta1/pkg/services/ngalert/notifier/channels/pagerduty.go#L84

**Comment by eraac** [comment]
We have just move to the new alerting system, and all the custom_data previously push into the PagerDuty payload are now missing, causing a lot of issues with your current PagerDuty configuration ... 

Previously all the tags was passed to PagerDuty [code](https://github.com/grafana/grafana/blob/main/pkg/services/alerting/notifiers/pagerduty.go#L169) and also the state [code](https://github.com/grafana/grafana/blob/main/pkg/services/alerting/notifiers/pagerduty.go#L116)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
